trajopt/include/trajopt/kinematic_terms.hpp

```diff
@@ -409,4 +409,84 @@ struct TimeCostJacCalculator : sco::MatrixOfVector
   Eigen::MatrixXd operator()(const Eigen::VectorXd& var_vals) const override;
 };
 
+/** @brief Cost calculator for evaluating the cost of a singularity, in the form 1.0 / (smallest_sv + lambda) */
+struct AvoidSingularityCostCalculator : sco::VectorOfVector
```

Comment:
I recommend changing this to AvoidSingularityErrCalculator to be consistent with naming convention. Having cost in the name could be deceiving since this could be used as either a cost or constraint.